test/runnable_cxx/test22287.d

```diff
@@ -0,0 +1,45 @@
+// EXTRA_CPP_SOURCES: cpp22287.cpp
+
+extern(C++):
+
+class A
+{
+    ~this();
+    int f1() const;
+
+    int i;
+}
+
+interface I
+{
+    int f2() const;
+}
+
+class B : A, I
+{
+    override int f1() const;
+    override int f2() const;
+    int f3() const;
+}
+
+class C : B
+{
+    override int f1() const;
+    override int f2() const;
+    override int f3() const;
+}
+
+extern(D) void main()
+{
+    B b = new B();
```

Comment:
To fully check C++ interoperability in this regard, it would probably make sense to do the allocation on one side, and the call on the other side. E.g., exposing some factories in D, like returning a new `C` instance typed as `{C,B,I}` (thus testing vptrs and vtables set up by D), calling them in C++ and then checking the calls. And then doing it the other way around, instantiation on the C++ side and calls in D.

Reply:
That is a good idea. It looks like I found a new issue now by extending the test. I will look into it further and update the pull request. Maybe I just made a mistake in the test.